components/fxa-client/src/internal/oauth.rs

```diff
@@ -383,16 +381,12 @@ impl FirefoxAccount {
             },
             None => None,
         };
-        self.state.refresh_token = Some(RefreshToken {
-            token: new_refresh_token,
-            scopes: resp.scope.split(' ').map(ToString::to_string).collect(),
-        });
         // In order to keep 1 and only 1 refresh token alive per client instance,
         // we also destroy the existing refresh token.
         if let Some(ref refresh_token) = old_refresh_token {
             if let Err(err) = self
                 .client
-                .destroy_refresh_token(&self.state.config, &refresh_token.token)
+                .destroy_refresh_token(self.state.config(), &refresh_token.token)
```

Comment:
Some of this is like the failure to register cases above. The comment implies it's important to only have 1 refresh token alive per client, but what if this fails? If it's important, shouldn't our state carry some kind of "pending operations" information so we can retry important things later?

It also seems like whether that's done or not, the state should maybe be responsible for trying this. IIUC, this code is effectively replacing the refresh token help by the state with another - ISTM that maybe this state setter should be the one to destroy the token being replaced?